### PR TITLE
Add Ubuntu Based DIND Agent

### PR DESCRIPTION
**Pull request by colin-msphere**
Add a docker-in-docker agent based on Ubuntu. This provides a typical glibc environment, whereas alpine uses musl.

### CODE REVIEW

**Comment by colin-msphere** [comment]
retest this please

**Comment by ghost** [comment]
Please rename `Dockerfile` to `Dockerfile.alpine`, switch to spaces instead of tabs, and align all backspaces in the Dockerfile. We'll also want to add/modify a build script to handle `Dockerfile.*`.

File: Dockerfile.ubuntu [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+       ca-certificates     \
+       ca-certificates-java    \
+       curl                \
+       git                 \
+       iptables            \
+       jq                  \
+       lvm2                \
+       lxc                 \
+       openjdk-8-jre-headless  \
+       unzip               \
```
Comment (sschneid) [inline Dockerfile.ubuntu]: nit: can these `\`s be aligned?

**Comment by ghost** [comment]
Is there a build script or build script changes that can be committed as part of this PR as well?

**Comment by colin-msphere** [comment]
@rji - there are no build scripts currently in this repo. Would you suggest a script that builds everything, a script per image, or an update to the README with build instructions?

**Comment by ghost** [comment]
A single script that builds everything.

File: Dockerfile.ubuntu [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+       bzip2                \
+       ca-certificates      \
+       ca-certificates-java \
+       curl                 \
+       git                  \
+       iptables             \
+       jq                   \
+       lvm2                 \
+       lxc                  \
+       openjdk-8-jre-headless  \
```
Comment (ghost) [inline Dockerfile.ubuntu:19]: nit: align `\`s

**Comment by ghost** [comment]
Could we make the script also do a `docker login` and `docker push`?

**Comment by colin-msphere** [comment]
I am adding a `push` piece, but I would like to leave `login` out and let the user handle that.

**Comment by ghost** [comment]
Good point, that sounds good to me. Thanks @colin-msphere!

**Comment by ghost** [comment]
Looking at the build script and how it's tagging and pushing images, this appears to tag the images using a convention like `mesosphere/jenkins-dind:alpine` and `mesosphere/jenkins-dind:ubuntu`. However, up to this point we've tagged the images with a version number and we have a CI job that builds and pushes the images to Docker Hub when a new tag is pushed to this repo.

I'm not sure that we can safely abandon our previous versioning scheme since it could lead to inconsistencies in users' environments (for example: using an older, cached image on one machine and pulling down a newer version on a machine where the image isn't already cached). I also don't think it's wise to recommend that users force pull this image every time a CI build needs to take place.

**Pull request merged**